api/src/pcapi/core/finance/api.py

```diff
@@ -2776,6 +2776,116 @@ def create_overpayment_finance_incident(
     return incident
 
 
+def create_overpayment_finance_incident_collective_booking(
+    booking: educational_models.CollectiveBooking,
+    origin: str,
+) -> models.FinanceIncident:
+    incident = models.FinanceIncident(
+        kind=models.IncidentType.OVERPAYMENT,
+        status=models.IncidentStatus.CREATED,
+        venueId=booking.venueId,
+        details={
+            "origin": origin,
+            "author": current_user.full_name,
+            "createdAt": datetime.datetime.utcnow().isoformat(),
+        },
+    )
+    db.session.add(incident)
+    db.session.flush()
+
+    booking_finance_incident = models.BookingFinanceIncident(
+        collectiveBookingId=booking.id,
+        incidentId=incident.id,
+        newTotalAmount=0,
+    )
+    db.session.add(booking_finance_incident)
+
+    history_api.add_action(
+        history_models.ActionType.FINANCE_INCIDENT_CREATED,
+        author=current_user,
+        finance_incident=incident,
+        comment=origin,
+    )
+
+    db.session.commit()
+
+    return incident
+
+
+def create_finance_commercial_gesture(
+    bookings: list[bookings_models.Booking],
+    amount: float,
+    origin: str,
+) -> models.FinanceIncident:
+    incident = models.FinanceIncident(
+        kind=models.IncidentType.COMMERCIAL_GESTURE,
+        status=models.IncidentStatus.CREATED,
+        venueId=bookings[0].venueId,
+        details={
+            "origin": origin,
+            "author": current_user.full_name,
+            "createdAt": datetime.datetime.utcnow().isoformat(),
+        },
+    )
+    db.session.add(incident)
+    db.session.flush()
+
+    booking_finance_incidents_to_create = []
+    for booking in bookings:
+        # all bookings in a commercial gesture must be from the same stock → they all have the same amount
+        new_total_amount = (amount / sum(booking.quantity for booking in bookings)) * booking.quantity
```

Comment:
`for booking in bookings` à l'intérieur d'une boucle `for booking in bookings` prête à confusion pour comprendre quel `booking` est utilisé à chaque endroit sur la ligne ! (certes, la portée est connue, mais on peut faire plus clair :slightly_smiling_face: )